### ISSUE TITLE
Vendor doesnt work well with Joypad

### ISSUE THREAD

**Issue by chameleonkid** [chameleonkid/Secrets#50]
Buttons clicked dont display the right Infotext.
Close-Button cant be clicked.

**Comment by itsschwer** [comment]
The issue seems to stem from how Unity's UI event system works, and isn't limited to joypads.

The event system only listens for input named `"Submit"` under the Input Manager. The desired joypad input, however, is named `"Interact"`. This can be replicated using keyboard by comparing the behaviour of inputting `e` (`"Interact"`) against the behaviour of inputting `enter`/`return` (`"Submit"`).

I'm still looking into why `"Interact"` is leading to any behaviour at all, since afaik, it's unexpected. In the meantime, it might be worth re-planning out and organising the Input Manager, @chameleonkid?

**Comment by itsschwer** [comment]
I've figured out the unexpected `"Interact"` behaviour. The vendor panel is activated using the `"Interact"` input, which did not check whether or not the vendor panel is already active before refreshing itself (where it calls `EventSystemSelectDefault();`), making it look like the `"Interact"` button was actually interacting with the item slots. In actuality, the `"Interact"` input was read by `Vendor`. This has been addressed in commit 5d45913.

`"Interact"` now should lead to no behaviour when the vendor panel is open, as expected (but perhaps not as desired).
